### PR TITLE
Gradient Ascent update for actors in Reinforcement learning.

### PR DESCRIPTION
**Pull request by mirraaj**
This performs Gradient Ascent for actor networks in Reinforcement learning. This optimiser can directly be used by the file. Since this is a very basic fine, I do not think we need any test file to test this.

### CODE REVIEW

**Comment by luffy1996** [comment]
Dear @zoq, @rcurtin  you might want to have a look at this. This implements gradient ascent for actor learners in reinforcement learning. I guess we do not need any test for this optimiser . Please provide me your inputs. Thanks

**Comment by luffy1996** [comment]
Hi @sourabhvarshney111 . Gradient ascent are procedures to increase the performance of actor/policy networks in reinforcement learning. These agents learn to play better by increasing their parameters from gradient ascent approach. I think gradient ascent is not used in machine learning/ deep learning because the sole purpose of these algorithms is minimisation of cost function. However for policy networks one needs to maximise the cost function to increase the performance of the networks. I hope I solved your doubt to some extent. Thanks

**Comment by sourabhvarshney111** [comment]
Thanks @luffy1996. I am clear now about the idea behind the gradient ascent update. I will learn more on this on my own. Thanks again : )

File: src/mlpack/core/optimizers/sga/sga_update.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#ifndef MLPACK_CORE_OPTIMIZERS_SGA_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_SGA_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Update policy for Stochastic Gradient Ascent (SGA).
```
Comment (sourabhvarshney111) [inline src/mlpack/core/optimizers/sga/sga_update.hpp]: Can you please add the references?

Comment (luffy1996) [reply src/mlpack/core/optimizers/sga/sga_update.hpp]: I think its not needed here. Its too trivial to be tested . You may find a similar implementation at https://github.com/mlpack/mlpack/blob/master/src/mlpack/core/optimizers/sgd/update_policies/vanilla_update.hpp . Let the reviewers @rcurtin , @zoq decide if we need a test for this file. However if you want to add a test, feel free to do so. Thanks

Comment (luffy1996) [reply src/mlpack/core/optimizers/sga/sga_update.hpp]: For the reference Part , I will do so as soon as possible. Thanks

Comment (sourabhvarshney111) [reply src/mlpack/core/optimizers/sga/sga_update.hpp]: Ok let @zoq @rcurtin decide about the tests. Rest after the reference, I will look to see this merged soon. : )

Comment (luffy1996) [reply src/mlpack/core/optimizers/sga/sga_update.hpp]: @sourabhvarshney111 I think the following commit might be more clear to you. Thanks

File: src/mlpack/core/optimizers/sga/sga_update.hpp [hunk 2]
```diff
@@ -18,7 +18,15 @@ namespace mlpack {
...
 /**
- * Update policy for Stochastic Gradient Ascent (SGA).
+ * Update policy for Gradient Ascent.
+ * @article{Kingma2014,
+ *   author  = {R.S. Sutton, David McAllester, Satinder Singh, Yishay Mansour},
+ *   title   = {Policy Gradient Methods for Reinforcement Learning with 
+                Function Approximation },
+ *   journal = {NIPS},
+ *   year    = {2000},
+ *   url     = {https://goo.gl/UwfMnd}
```
Comment (sourabhvarshney111) [inline src/mlpack/core/optimizers/sga/sga_update.hpp]: I think we can add the official url:
https://papers.nips.cc/paper/1713-policy-gradient-methods-for-reinforcement-learning-with-function-approximation.pdf
Check whether this is official or not please. I have just tried to find that out.

File: src/mlpack/core/optimizers/sga/sga_update.hpp [hunk 3]
```diff
@@ -25,7 +25,7 @@ namespace optimization {
                 Function Approximation },
  *   journal = {NIPS},
  *   year    = {2000},
- *   url     = {https://goo.gl/UwfMnd}
+ *   url     = {https://goo.gl/2Gtmqu}
```
Comment (sourabhvarshney111) [inline src/mlpack/core/optimizers/sga/sga_update.hpp]: I don't know whether we allow google shorteners or not. I was searching for it in the design wiki. I think @zoq @rcurtin would give you better idea on that.

**Comment by luffy1996** [comment]
@zoq, Please provide me how to integrate this with existing codebase. I guess actor only or actor-critic algorithms are not currently supported in mlpack. For asynchronous method https://github.com/mlpack/mlpack/tree/master/src/mlpack/methods/reinforcement_learning/worker only one step q learning, n step q learning, one step sarsa seems to be implemented. How should I proceed here? Thanks

File: src/mlpack/core/optimizers/sga/sga_update.hpp (hunk 3 above)
Comment (luffy1996) [reply src/mlpack/core/optimizers/sga/sga_update.hpp]: Dear @zoq , @rcurtin The link to the paper is long. The link is https://papers.nips.cc/paper/1713-policy-gradient-methods-for-reinforcement-learning-with-function-approximation.pdf .Currently I am using a google shortner.  How should I reference it here . Thanks

**Comment by luffy1996** [comment]
Dear @zoq , @rcurtin I am going with #1328 implementation to test my gradient ascent. The previous PR is hence closed. Thanks

**Pull request closed without merging**